### PR TITLE
Alter so labels with numbers and underscores dont get needless backticks

### PR DESCRIPTION
**Pull request by anderson4j**
Before if for example had relationship rel_123, we autocompleted `MATCH (n)-[:rel^]` to
` MATCH (n)-[:`\``rel_123`\``] `
Now instead we dont backtick if we have underscore and numbers in label names/rel names/properties and just give
`MATCH (n)-[:rel_123]`

### CODE REVIEW

File: packages/language-support/src/autocompletion/completionCoreCompletions.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ import { CompletionItem, Neo4jFunction, Neo4jProcedure } from '../types';
 const uniq = <T>(arr: T[]) => Array.from(new Set(arr));
 
 function backtickIfNeeded(e: string): string | undefined {
-  if (/\s/.test(e) || /[^a-zA-Z]/.test(e)) {
+  if (/\s/.test(e) || /[^a-zA-Z0-9_]/.test(e)) {
```
Comment (ncordon) [inline packages/language-support/src/autocompletion/completionCoreCompletions.ts:35]: There's an edge case we are not really handling well here: a label, rel type or property key that starts with a number still needs backticks

Comment (anderson4j) [reply packages/language-support/src/autocompletion/completionCoreCompletions.ts:35]: Also checked with chatGPT and apparently non-english letters is okay anywhere in the names, will fix both at the same time

**Review by ncordon — CHANGES_REQUESTED** [review]
Looks good to me but I'd like to see a case where the label / rel type starts by underscore added to the test cases

File: packages/language-support/src/autocompletion/completionCoreCompletions.ts (hunk 1 above)
Comment (ncordon) [reply packages/language-support/src/autocompletion/completionCoreCompletions.ts]: I've just checked with an instance of the database to be sure. Please check always with the database itself because you've seen ChatGPT already give hallucinations regarding neo4j documentation for procedures.

File: packages/language-support/src/autocompletion/completionCoreCompletions.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ import { CompletionItem, Neo4jFunction, Neo4jProcedure } from '../types';
 const uniq = <T>(arr: T[]) => Array.from(new Set(arr));
 
 function backtickIfNeeded(e: string): string | undefined {
-  if (/\s/.test(e) || /[^a-zA-Z]/.test(e)) {
+  if (/[^\p{L}\p{N}_]/u.test(e) || /[^\p{L}_]/u.test(e[0])) {
```
Comment (ncordon) [inline packages/language-support/src/autocompletion/completionCoreCompletions.ts:35]: We should check first the string is not empty:

```typescript
if (/[^\p{L}\p{N}_]/u.test(e) || (e.length > 0 && /[^\p{L}_]/u.test(e[0]))
```

File: packages/language-support/src/autocompletion/completionCoreCompletions.ts (hunk 1 above)
Comment (anderson4j) [reply packages/language-support/src/autocompletion/completionCoreCompletions.ts]: I do indeed know about hallucinations and have also checked, no need to worry

**Review by ncordon — APPROVED** [review]
